Libraries/plist/Tools/plutil.cpp

```diff
@@ -594,31 +640,42 @@ Modify(Filesystem *filesystem, Options const &options, std::string const &file,
         } while (end != std::string::npos);
     }
 
-    /* Find output format. */
-    plist::Format::Any out = format;
-    if (options.convert()) {
-        switch (*options.convert()) {
-            case plist::Format::Type::Binary: {
-                plist::Format::Binary binary = plist::Format::Binary::Create();
-                out = plist::Format::Any::Create<plist::Format::Binary>(binary);
-                break;
-            }
-            case plist::Format::Type::XML: {
-                plist::Format::XML xml = plist::Format::XML::Create(plist::Format::Encoding::UTF8);
-                out = plist::Format::Any::Create<plist::Format::XML>(xml);
-                break;
+    /* Convert to desired format. */
+    std::pair<std::unique_ptr<std::vector<uint8_t>>, std::string> serialize;
+    if (ext::optional<Options::Format> format = options.convert()) {
+        if (ext::optional<plist::Format::Type> type = format->type()) {
+            switch (*type) {
+                case plist::Format::Type::Binary: {
+                    plist::Format::Binary binary = plist::Format::Binary::Create();
+                    serialize = plist::Format::Binary::Serialize(writeObject, binary);
+                    break;
+                }
+                case plist::Format::Type::XML: {
+                    plist::Format::XML xml = plist::Format::XML::Create(plist::Format::Encoding::UTF8);
+                    serialize = plist::Format::XML::Serialize(writeObject, xml);
+                    break;
+                }
+                case plist::Format::Type::ASCII: {
+                    plist::Format::ASCII ascii = plist::Format::ASCII::Create(false, plist::Format::Encoding::UTF8);
+                    serialize = plist::Format::ASCII::Serialize(writeObject, ascii);
+                    break;
+                }
+                default: abort();
             }
-            case plist::Format::Type::ASCII: {
-                plist::Format::ASCII ascii = plist::Format::ASCII::Create(false, plist::Format::Encoding::UTF8);
-                out = plist::Format::Any::Create<plist::Format::ASCII>(ascii);
-                break;
+        } else if (ext::optional<Options::Format::Additional> additional = format->additional()) {
```

Comment:
See the discussion on #165 — in short, `plist::Format::Type` doesn't include JSON (and I don't want to add it).